### ISSUE TITLE
Add filter options to the `page_view` `AutoCapture`

### ISSUE THREAD

**Issue by valkum** [statsig-io/js-client-monorepo#14]
Currently, the `page_view` `AutoCapture` tracks each change to the `navigration.href` as a `page_view`. If you track some React state in your URL (such as modal states, etc.), this creates a `page_view` event for each state change. We would like to be able to filter out specific query params that can change and should not trigger a `page_view`.

To implement this, we need some way to pass options to the AutoCapture feature.
The actual check for ignored changed parts of a URL should be easy to add to https://github.com/statsig-io/js-client-monorepo/blob/main/packages/web-analytics/src/AutoCapture.ts#L180

**Comment by daniel-statsig** [comment]
Sorry this took so long. A solution for this has now been shipped in 3.7.0 (https://github.com/statsig-io/js-client-monorepo/releases/tag/3.7.0).

You can no use the new `eventFilterFunc` to do something like:

```ts
const seen: Record<string, boolean> = {};

const eventFilterFunc = (event: AutoCaptureEvent) => {
  if (event.eventName !== 'auto_capture::page_view') {
    return true;
  }

  if (event.metadata && typeof event.metadata['page_url'] === 'string') {
    const url = new URL(event.metadata['page_url']);
    if (seen[url.pathname]) {
      return false;
    }

    seen[url.pathname] = true;
  }

  return true;
};

const client = new StatsigClient(
  DEMO_CLIENT_KEY,
  { userID: 'a-user' },
  {
    plugins: [
      new StatsigAutoCapturePlugin({
        eventFilterFunc,
      }),
    ],
  },
);
```

**Comment by valkum** [comment]
Thanks!
